lib/straight-server/gateway.rb

```diff
@@ -240,6 +240,11 @@ def send_callback_http_request(order, delay: 5)
         callback_data = order.callback_data ? "&callback_data=#{CGI.escape(order.callback_data)}" : ''
         begin
           uri = URI.parse("#{url}#{url.include?('?') ? '&' : '?'}#{order.to_http_params}#{callback_data}")
+          unless uri.kind_of?(URI::HTTP) || uri.kind_of?(URI::HTTPS)
+            StraightServer.logger.info "Looks like callback_url - #{url} is invalid\nProceeding via http protocol "
+            url = 'http://' + url
+            uri = URI.parse("#{url}#{url.include?('?') ? '&' : '?'}#{order.to_http_params}#{callback_data}")
```

Comment:
I don't like this code duplication. You should implement this logic before the string N242. I recommend you use `URI.split`